core/src/autogluon/core/utils/utils.py

```diff
@@ -5,26 +5,12 @@
 from datetime import datetime
 
 import numpy as np
-import pandas as pd
-from pandas import DataFrame, Series
-from sklearn.model_selection import KFold, StratifiedKFold, RepeatedKFold, RepeatedStratifiedKFold, train_test_split
-
-from .constants import BINARY, REGRESSION, MULTICLASS, SOFTCLASS
-from ..metrics import accuracy, root_mean_squared_error, Scorer
+from pandas import DataFrame
+from sklearn.model_selection import KFold, StratifiedKFold, RepeatedKFold, RepeatedStratifiedKFold
```

Comment:
Is `sklearn` often used elsewhere in core? Otherwise could consider moving these functions to reduce one more dependency

Reply:
it's also used in grid_searcher (just ParameterGrid). However, the functions in `utils.py` are used across the core, so it's not easily extract-able.